src/tools/simeng/main.cc

```diff
@@ -41,38 +49,37 @@ int main(int argc, char** argv) {
   std::cout << "[SimEng] \tTest suite: " SIMENG_ENABLE_TESTS << std::endl;
   std::cout << std::endl;
 
-  // Create the instance of the core to be simulated
-  std::unique_ptr<simeng::CoreInstance> coreInstance;
-  std::string executablePath = "";
-  std::string configFilePath = "";
-  std::vector<std::string> executableArgs = {};
-
+  // Parse command line args
+  std::string executablePath = DEFAULT_STR;
+  std::vector<std::string> executableArgs;
   // Determine if a config file has been supplied.
   if (argc > 1) {
-    configFilePath = std::string(argv[1]);
+    // Set global config file to one at file path defined
+    Config::set(std::string(argv[1]));
+
     // Determine if an executable has been supplied
     if (argc > 2) {
       executablePath = std::string(argv[2]);
       // Create a vector of any potential executable arguments from their
       // relative position within the argv variable
-      char** startOfArgs = argv + 3;
       int numberofArgs = argc - 3;
       executableArgs =
-          std::vector<std::string>(startOfArgs, startOfArgs + numberofArgs);
+          std::vector<std::string>((argv + 3), (argv + 3) + numberofArgs);
     }
-    coreInstance = std::make_unique<simeng::CoreInstance>(
-        configFilePath, executablePath, executableArgs);
-  } else {
-    // Without a config file, no executable can be supplied so pass default
-    // (empty) values for executable information
-    coreInstance =
-        std::make_unique<simeng::CoreInstance>(executablePath, executableArgs);
-    configFilePath = "Default";
   }
 
-  // Replace empty executablePath string with more useful content for
-  // outputting
-  if (executablePath == "") executablePath = "Default";
+  // Create global memory
+  std::shared_ptr<simeng::memory::Mem> memory =
+      std::make_shared<simeng::memory::SimpleMem>(2684354560);  // 2.6 GiB
+
+  // Create the instance of the OS
+  simeng::kernel::SimOS simOS_kernel =
```

Comment:
`simOS_kernel` naming convention deviates from the rest of the codebase. I think `simOSKernel` is better suited. It's not perfect due to the adjacent uppercased (S and K) but could still be better than the current choice.

Reply:
I think even changin this to `kernel` would suffice?

Reply:
Yes, actually just `kernel` would be good.